File: app/src/main/java/com/visual/visualchart/MainActivity.java
package com.visual.visualchart;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;

import com.blankj.utilcode.util.GsonUtils;
import com.github.lzyzsd.jsbridge.BridgeHandler;
import com.github.lzyzsd.jsbridge.CallBackFunction;
import com.visual.visualchart.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private static String TAG = "MainActivity";
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.webView.registerHandler("dataWebToNative", new BridgeHandler() {
            @Override
            public void handler(String data, CallBackFunction function) {
                binding.text2.setText(GsonUtils.fromJson(data,DataPar.class).param);
                function.onCallBack("submitFromWeb exe, response data from Java");
            }
        });

        binding.btnSend.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                binding.webView.callHandler("dataNativeToWeb", GsonUtils.toJson(new DataPar("dataNativeToWeb")), new CallBackFunction() {
                    @Override
                    public void onCallBack(String data) {
                        binding.text2.setText(data);
                    }
                });
            }
        });
        binding.webView.loadUrl("file:///android_asset/demo.html");
    }

    public static class DataPar {
        private String param;

        public DataPar(String param) {
            this.param = param;
        }
    }
}

File: app/src/main/java/com/visual/visualchart/app/ui/main/view/MainActivity.kt
package com.visual.visualchart.app.ui.main.view

import com.visual.visualchart.app.base.BaseActivity
import com.visual.visualchart.app.ui.main.viewmodel.HomeViewModel
import android.os.Bundle
import com.blankj.utilcode.util.GsonUtils
import com.visual.visualchart.app.data.WebToNativeDataPar
import com.visual.visualchart.databinding.ActivityMainBinding

class MainActivity : BaseActivity<HomeViewModel, ActivityMainBinding>() {
    override fun initView(savedInstanceState: Bundle?) {
        mViewBind.webView.registerHandler("dataWebToNative") { data, function ->
            mViewBind.text2.text = GsonUtils.fromJson(data, WebToNativeDataPar::class.java).param
            function.onCallBack("submitFromWeb exe, response data from Java")
        }
        mViewBind.btnSend.setOnClickListener {
            mViewBind.webView.callHandler(
                "dataNativeToWeb",
                GsonUtils.toJson(WebToNativeDataPar("dataNativeToWeb"))
            ) { data -> mViewBind.text2.text = data }
        }
        mViewBind.webView.loadUrl("file:///android_asset/demo.html")
    }

    companion object {
        private const val TAG = "MainActivity"
    }
}